packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -1536,36 +1558,48 @@ class RectangularSliderTrackShape extends SliderTrackShape with BaseSliderTrackS
     // Assign the track segment paints, which are left: active, right: inactive,
     // but reversed for right to left text.
     final ColorTween activeTrackColorTween = ColorTween(begin: sliderTheme.disabledActiveTrackColor, end: sliderTheme.activeTrackColor);
+    final ColorTween secondaryValueTrackColorTween = ColorTween(begin: sliderTheme.disabledSecondaryValueTrackColor, end: sliderTheme.secondaryValueTrackColor);
     final ColorTween inactiveTrackColorTween = ColorTween(begin: sliderTheme.disabledInactiveTrackColor, end: sliderTheme.inactiveTrackColor);
     final Paint activePaint = Paint()..color = activeTrackColorTween.evaluate(enableAnimation);
+    final Paint secondaryValueTrackPaint = Paint()..color = secondaryValueTrackColorTween.evaluate(enableAnimation);
     final Paint inactivePaint = Paint()..color = inactiveTrackColorTween.evaluate(enableAnimation);
     Paint leftTrackPaint;
     Paint rightTrackPaint;
+    bool needPaintSecondaryValueTrack = false;
+
+    final Rect trackRect = getPreferredRect(
+      parentBox: parentBox,
+      offset: offset,
+      sliderTheme: sliderTheme,
+      isEnabled: isEnabled,
+      isDiscrete: isDiscrete,
+    );
+
     switch (textDirection) {
       case TextDirection.ltr:
         leftTrackPaint = activePaint;
         rightTrackPaint = inactivePaint;
+        needPaintSecondaryValueTrack = secondaryValueTrackRight != null && secondaryValueTrackRight > thumbCenter.dx;
         break;
       case TextDirection.rtl:
         leftTrackPaint = inactivePaint;
         rightTrackPaint = activePaint;
+        needPaintSecondaryValueTrack = secondaryValueTrackRight != null && secondaryValueTrackRight > trackRect.right - thumbCenter.dx;
         break;
     }
 
-    final Rect trackRect = getPreferredRect(
-      parentBox: parentBox,
-      offset: offset,
-      sliderTheme: sliderTheme,
-      isEnabled: isEnabled,
-      isDiscrete: isDiscrete,
-    );
-
     final Size thumbSize = sliderTheme.thumbShape.getPreferredSize(isEnabled, isDiscrete);
 
-    final Rect leftTrackSegment = Rect.fromLTRB(trackRect.left + trackRect.height / 2, trackRect.top, thumbCenter.dx - thumbSize.width / 2, trackRect.bottom);
```

Comment:
The `leftTrackSegment.left` should not plus `trackRect.height / 2`, it is probably a bug here.
Related commits: https://github.com/flutter/flutter/pull/31564/files#diff-71a1fc327f48d2a30bc5c79f290d5391R1154

Reply:
I changed the `leftTrackSegment.right` from `thumbCenter.dx - thumbSize.width / 2` to `thumbCenter.dx` .
There are two reasons:
1. If the slider's height > thumb's size, there will be a gap 
![image](https://user-images.githubusercontent.com/61075224/91041645-fb813380-e642-11ea-9d81-14d98b696aca.png)
2. The process is different with `RoundedRectSliderTrackShape`

Reply:
Track at #64696